plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/worker/ConsoleReportWorkAction.kt

```diff
@@ -0,0 +1,67 @@
+package org.jlleitschuh.gradle.ktlint.worker
+
+import com.pinterest.ktlint.core.LintError
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.logging.Logging
+import org.gradle.api.provider.Property
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import java.io.FileInputStream
+import java.io.ObjectInputStream
+
+@Suppress("UnstableApiUsage")
+internal abstract class ConsoleReportWorkAction : WorkAction<ConsoleReportWorkAction.ConsoleReportParameters> {
+
+    private val logger = Logging.getLogger("ktlint-console-report-worker")
+
+    override fun execute() {
+        val errors = loadErrors()
+
+        if (parameters.outputToConsole.getOrElse(false)) {
+            val verbose = parameters.verbose.get()
+            errors.forEach { lintErrorResult ->
+                val filePath = lintErrorResult.lintedFile.absolutePath
+                lintErrorResult
+                    .lintErrors
+                    .filter { !it.second }
+                    .forEach {
+                        it.first.lintError.logError(filePath, verbose)
+                    }
+            }
+        }
+
+        if (errors.any { it.lintErrors.any { !it.second } } &&
+            !parameters.ignoreFailures.getOrElse(false)
+        ) {
+            throw GradleException("KtLint found code style violations.")
```

Comment:
In the future, or now, including the path to the report would be nice here.

Reply:
Done via 0182240